drivers/overlay/overlay.go

```diff
@@ -942,6 +942,20 @@ func (d *Driver) create(id, parent string, opts *graphdriver.CreateOpts, disable
 		rootUID = int(st.UID())
 		rootGID = int(st.GID())
 	}
+
+	// --- Only access "dir" below.
+	d.locker.Lock(id)
+	defer d.locker.Unlock(id)
+
+	if _, err := system.Lstat(dir); err == nil {
+		logrus.Warnf("Trying to create a layer %#v while directory %q already exists; removing it first", id, dir)
```

Comment:
We default to showing warnings.  Is this something we want normal users to see by default?  Or should we drop this down to info?

Reply:
I was thinking that, this indicates an unexpected abort of a c/storage operation, and it is definitely something we want to have captured _in any logs at the default level_ for postmortems if this cleanup is insufficient and/or incorrect and harmful.

I’m not so sure that it needs to be printed by default on a TTY during interactive use; at that point, there are unlikely to be any logs recorded, and the thing either works or it doesn’t. OTOH, it’s also a situation we shouldn’t be regularly getting into, so I don’t think that a warning should _hurt_.

Alternatively, #1148 should make this case unreachable assuming there are no bugs — and in that case we either don’t need this PR, or we can happily leave it on a warning level because it is clearly very unexpected. If I’m reading @nalind’s comments right, it seems preferable to have both this PR and #1148.

Reply:
Ok leave it at Warning level.